rtems/RTEMSThread.java

```diff
@@ -15,6 +15,8 @@ public RTEMSThread(int priority, Thread.State state){
 		this();
 		this.currentPriority = this.realPriority = priority;
 		this.state = state;
+		//this.lockMutex = new ChainControl();
+		PriorityQueue<RTEMSThread> wait = null;
```

Comment:
This goes out of scope as soon as the constructor is finished. I think you mean to declare `PriorityQueue<RTEMSThread> wait` as the class instance variable and assign to it rather than the `PriorityQueue<Object> wait` declaration.